recovery.fstab

```diff
@@ -27,11 +27,11 @@
 
 # mount point	fstype		device			[device2] [length=]
 
-/dev/block/platform/msm_sdcc.1/by-name/system       /system         ext4    ro,barrier=1                                                    wait
-/dev/block/platform/msm_sdcc.1/by-name/cache        /cache          ext4    noatime,nosuid,nodev,barrier=1,data=ordered                     wait,check
-/dev/block/platform/msm_sdcc.1/by-name/userdata     /data           ext4    noatime,nosuid,nodev,barrier=1,data=ordered,noauto_da_alloc     wait,check
+/dev/block/platform/msm_sdcc.1/by-name/system       /system         ext4    ro,barrier=1,discard                                            wait
+/dev/block/platform/msm_sdcc.1/by-name/cache        /cache          ext4    nosuid,nodev,discard                                            wait,check
+/dev/block/platform/msm_sdcc.1/by-name/userdata     /data           ext4    nosuid,nodev,barrier=1,noauto_da_alloc,discard                  wait,check,encryptable=footer
```

Comment:
Why remove noatime? I also noticed that the regular (non-recovery) mounts for the z3c are using relatime rather than noatime, but conventional wisdom seems to suggest that using noatime is better for performance and reduces flash writes.

And why remove barrier=1 on /cache? barrier=1 seems to be important when using a journalling filesystem (like ext4)

data=ordered is the default if not specified, so that seems ok.